utils/migration_tools/from_another_project/runs_migrator.py

```diff
@@ -0,0 +1,263 @@
+)
+logging.basicConfig(
+    filename=log_filename,
+    filemode="a",
+    format="%(asctime)s %(name)s %(levelname)s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+    level=logging.INFO,
+    force=True,
+)
+
+print(f"Logs available at {log_filename}")
+
+logging.getLogger("neptune.internal.operation_processors.async_operation_processor").setLevel(
+    logging.CRITICAL
+)
+
+# %%
+projects = management.get_project_list()
+
+if from_project not in projects:
+    logging.error(f"Project {from_project} does not exist. Please check project name")
+elif to_project not in projects:
+    logging.error(f"Project {to_project} does not exist. Please check project name")
+else:
+    logging.info(f"Copying from {from_project} to {to_project}")
+
+# %% Get list of runs to be copied
+with neptune.init_project(
+    project=from_project,
+    mode="read-only",
+) as neptune_from_project:
+    to_copy = neptune_from_project.fetch_runs_table(columns=[]).to_pandas()["sys/id"].values
+
+logging.info(f"{len(to_copy)} runs found")
+
+
+# %%
+def flatten_namespaces(
+    dictionary: dict, prefix: Optional[list] = None, result: Optional[list] = None
+) -> list:
+    if prefix is None:
+        prefix = []
+    if result is None:
+        result = []
+
+    for k, v in dictionary.items():
+        if isinstance(v, dict):
+            flatten_namespaces(v, prefix + [k], result)
+        elif prefix_str := "/".join(prefix):
+            result.append(f"{prefix_str}/{k}")
+        else:
+            result.append(k)
+    return result
+
+
+# %%
+
+READ_ONLY_NAMESPACES = [
+    "sys/creation_time",
+    "sys/id",
+    "sys/modification_time",
+    "sys/monitoring_time",
+    "sys/owner",
+    "sys/ping_time",
+    "sys/running_time",
+    "sys/size",
+    "sys/trashed",
+]
+
+MAPPED_NAMESPACES = {
+    namespace: namespace.replace("sys", "old_sys") for namespace in READ_ONLY_NAMESPACES
+}
+
+UNFETCHABLE_NAMESPACES = [
+    "sys/state",
+    "sys/custom_run_id",  # This is being set separately
+    "source_code/git",
+]
+
+
+# %%
+for from_run_id in tqdm(to_copy):
+    with neptune.init_run(
+        project=from_project,
+        with_id=from_run_id,
+        mode="read-only",
+    ) as from_run:
+        custom_run_id = None
+
+        with contextlib.suppress(MissingFieldException):
+            custom_run_id = from_run["sys/custom_run_id"].fetch()
+
+        with neptune.init_run(
+            project=to_project,
+            custom_run_id=custom_run_id or None,
+            capture_hardware_metrics=False,
+            capture_stderr=False,
+            capture_traceback=False,
+            git_ref=GitRef.DISABLED,
+            source_files=[],
+        ) as to_run:
+            to_run_id = to_run["sys/id"].fetch()
+            logging.info(f"Copying {from_project}/{from_run_id} to {to_project}/{to_run_id}")
+
+            namespaces = flatten_namespaces(from_run.get_structure())
+
+            for namespace in namespaces:
```

Comment:
Probably add warnings for when copying from a run's namespace fails.

Reply:
Already handled here: https://github.com/neptune-ai/examples/pull/251/files#diff-d07e1d383cb63514632d1083ccc4b1fcb131d9ca42125ead53a6d4bcbd3da6e8R215-R217